components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/JWTTokenIssuer.java

```diff
@@ -70,25 +84,29 @@
     private static final String SHA256_WITH_RSA = "SHA256withRSA";
     private static final String SHA384_WITH_RSA = "SHA384withRSA";
     private static final String SHA512_WITH_RSA = "SHA512withRSA";
-    private static final String PS256 = "PS256";
     private static final String SHA256_WITH_HMAC = "SHA256withHMAC";
     private static final String SHA384_WITH_HMAC = "SHA384withHMAC";
     private static final String SHA512_WITH_HMAC = "SHA512withHMAC";
     private static final String SHA256_WITH_EC = "SHA256withEC";
     private static final String SHA384_WITH_EC = "SHA384withEC";
     private static final String SHA512_WITH_EC = "SHA512withEC";
 
+    private static final String KEY_STORE_EXTENSION = ".jks";
     private static final String AUTHORIZATION_PARTY = "azp";
-    private static final String CLIENT_ID = "client_id";
     private static final String AUDIENCE = "aud";
     private static final String SCOPE = "scope";
     private static final String TOKEN_BINDING_REF = "binding_ref";
     private static final String TOKEN_BINDING_TYPE = "binding_type";
-    private static final String DEFAULT_TYP_HEADER_VALUE = "at+jwt";
-    private static final String CNF = "cnf";
+
+    // To keep track of the expiry time provided in the original jwt assertion, when JWT grant type is used.
+    private static final String EXPIRY_TIME_JWT = "EXPIRY_TIME_JWT";
 
     private static final Log log = LogFactory.getLog(JWTTokenIssuer.class);
     private static final String INBOUND_AUTH2_TYPE = "oauth2";
+
+    // We are keeping a private key map which will have private key for each tenant domain. We are keeping this as a
+    // static Map since then we don't need to read the key from keystore every time.
+    private static Map<Integer, Key> privateKeys = new ConcurrentHashMap<>();
```

Comment:
the key manager implementation should have the in-memory static map, we should you that to get the keystores other than duplicating in this class

Reply:
This was an issue occured during merge. It has been fixed.